firestarter/workflows/ci/bash_runner.py

```diff
@@ -0,0 +1,57 @@
+"""This module contains functions for working with dagger containers."""
+import tempfile
+from typing import List, TextIO
+
+def create_script_from_commands(commands: List[str], file: TextIO) -> None:
+    """Builds a bash script from a list of commands.
+
+    Args:
+        commands: A list of strings representing commands.
+        file: A file object where the script will be written.
+    """
+    script: str = "\n".join(["#!/bin/bash"] + commands)
+
+    file.write(script.encode())
+    file.close()
```

Comment:
I know this was ported from older code but receiving a file descriptor and closing it inside the function call seems a bit off 😅

Reply:
I agree. This code has been refactored